core/src/test/java/org/infinispan/test/AbstractInfinispanTest.java

```diff
@@ -72,6 +92,67 @@
 @TestSelector(interceptors = AbstractInfinispanTest.OrderByInstance.class)
 public abstract class AbstractInfinispanTest {
 
+   static {
```

Comment:
One downside to putting this in `AbstractInfinispanTest` is that it can only be used in modules using TestNG.
I guess you only care about core tests at this stage, but it's something to consider.

Reply:
Yeah, I agree - I was going to move this elsewhere at some point. Wasn't quite sure yet though, so I just left it here to at least test core.

Reply:
IMO it should be a separate class that `AbstractInfinispanTest` invokes in a static initializer, no point in putting all this code in `AbstractInfinispanTest` now just to move it later.

Reply:
Sure, I can try that out :+1:

Reply:
The only problem about it being in this class is we can't reference the classes in other modules (unless I did something like a ServiceLoader to find things to register based on what modules are in the class path).